--- ../LINESEL.ZD2.code.asm	2025-06-21 21:18:29.115870357 -0600
+++ LINESEL.asm	2025-06-23 18:18:03.748626382 -0600
@@ -17,7 +17,7 @@
            CALLP.S2      __call_stub,B3
  ||        MV.L2         B0,B31
            MV.L2X        A4,B0
-    [ B0]  BNOP.S1       $C$L1,5
+    [ B0]  BNOP.S1       $C$L1_1,5
            ;.fphead       n, l, W, BU, nobr, nosat, 0101111b
            LDW.D2T2      *+B5[3],B0
            MVK.S1        0x6666,A6
@@ -25,12 +25,12 @@
            MV.L1         A0,A4
            NOP           1
            B.S2          B0
-           ADDKPC.S2     $C$RL1,B3,4
-;$C$RL1_1:
-$C$RL1:
-           BNOP.S1       $C$L2,5
-;$C$L1_1:
-$C$L1:
+           ADDKPC.S2     $C$RL1_1,B3,4
+$C$RL1_1:
+;$C$RL1:
+           BNOP.S1       $C$L2_1,5
+$C$L1_1:
+;$C$L1:
            LDW.D2T2      *B5[3],B0
            ;.fphead       n, l, W, BU, br, nosat, 1001000b
            LDW.D2T2      *++B15[2],B3
@@ -41,8 +41,8 @@
            SHL.S2        B4,0x17,B4
            MV.L1         A0,A4
            NOP           3
-;$C$L2_1:
-$C$L2:
+$C$L2_1:
+;$C$L2:
            LDW.D2T2      *++B15[2],B3
            NOP           4
            ;.fphead       n, l, W, BU, nobr, nosat, 1110000b
@@ -80,7 +80,7 @@
            MVK.S1        20,A4
            MV.L1X        B6,A6
            ADD.L1        A0,A4,A4
-           NOP           2
+           ;NOP           2
 
 	.def Fx_SFX_LineSel_OutLvl_edit
 	.align 32
@@ -125,9 +125,9 @@
            MV.L2         B4,B1
            ADD.L2X       B4,A0,B4
            LDW.D2T2      *B4[0],B0
-           MVK.S2        0x026c,B4
+           MVKL.S2       _Fx_SFX_LineSel_Coe,B4
 	; 0000010c  00000109 R_C6000_ABS_L16        8000026c   _Fx_SFX_LineSel_Coe + 0
-           MVKH.S2       0x80000000,B4
+           MVKH.S2       _Fx_SFX_LineSel_Coe,B4
 	; 00000110  0000010a R_C6000_ABS_H16        8000026c   _Fx_SFX_LineSel_Coe + 0
            MV.L1         A4,A10
            LDW.D1T1      *A4[1],A4
@@ -148,77 +148,81 @@
 	.def Fx_SFX_LineSel
 	.align 32
 Fx_SFX_LineSel:
-           LDW.D2T2      *+B4[7],B5
-           LDW.D2T2      *+B4[1],B20
-           LDW.D2T2      *+B4[8],B21
-           LDW.D2T1      *+B4[2],A5
-           MVK.L2        0,B19
-           MVK.L2        0,B9
-           MVK.S1        16,A0
-           LDW.D1T2      *A4[1],B4
-           ;.fphead       n, l, W, BU, nobr, nosat, 1000000b
- ||        SET.S2        B9,23,29,B9
- ||        MVK.L2        2,B0
-           MV.L2         B21,B6
- ||        ADD.S2        B19,B20,B8
- ||        LDW.D2T2      *B5[0],B7
- ||        ADD.L1        A0,-1,A0
- ||        ADD.S1X       A5,B19,A4
-;$C$L1_5:
-$C$L1:
-;$C$L2_5:
-$C$L2:
-           LDW.D2T2      *B6++[1],B5
-           NOP           4
-           STW.D2T2      B5,*+B7[0]
-           ;.fphead       n, l, W, BU, nobr, nosat, 0101010b
-           LDW.D2T2      *+B4[5],B17
-           LDW.D2T2      *+B8[0],B5
-           NOP           3
-           LDW.D2T2      *B4[0],B16
-           MPYSP.M2      B5,B17,B18
-           NOP           2
-           LDW.D2T2      *B4[4],B17
-           MPYSP.M2      B16,B18,B16
-           NOP           3
-           ;.fphead       n, h, W, BU, nobr, nosat, 0010100b
-           MPYSP.M2      B17,B16,B16
-           NOP           3
-           STW.D2T2      B16,*B8++[1]
-           LDW.D2T2      *B4[6],B16
-           LDW.D2T2      *B4[0],B17
-           NOP           4
-           SUBSP.L2      B9,B17,B16
- ||        MPYSP.M2      B5,B16,B5
-           ;.fphead       n, h, W, BU, nobr, nosat, 0001000b
-           NOP           2
-           LDW.D2T2      *+B4[4],B17
-           MPYSP.M2      B16,B5,B5
-           NOP           2
-           LDW.D1T1      *A4[0],A3
-           MPYSP.M2      B17,B5,B5
-           NOP           2
-    [ A0]  BDEC.S1       $C$L1,A0
-           ;.fphead       n, l, W, BU, nobr, nosat, 0001000b
-           ADDSP.L2X     B5,A3,B5
-           NOP           3
-           STW.D1T2      B5,*A4++[1]
-           ADD.L2        B0,-1,B0
- ||        MVK.S1        16,A3
- ||        ADDK.S2       64,B19
- ||        MV.D2         B21,B6
-           ADD.L1        A3,-1,A0
- ||        ADD.L2        B19,B20,B8
- || [ B0]  B.S1          $C$L1
-           ;.fphead       n, l, W, BU, nobr, nosat, 0010110b
-    [!B0]  BNOP.S2       B3,4
+           LDW.D2T2      *+B4[7],B5	; pointer to 'something' structure
+           LDW.D2T2      *+B4[1],B20	; base of 'effect' output buffer
+           LDW.D2T2      *+B4[8],B21	; base of 'effect' input buffer
+           LDW.D2T1      *+B4[2],A5	; base of 'final' output buffer
+
+           MVK.L2        0,B19		; zero offset into buffers
+           MVK.L2        0,B9		;
+           MVK.S1        16,A0		; prep inner loop x16
+           LDW.D1T2      *A4[1],B4	; base of 'parameters'
+ ||        SET.S2        B9,23,29,B9	; set B9 = 1.0 float
+ ||        MVK.L2        2,B0		; prep outer loop x2
+
+           MV.L2         B21,B6		; pointer to 'effect' input
+ ||        ADD.S2        B19,B20,B8	; add offset, pointer to 'effects' output
+ ||        LDW.D2T2      *B5[0],B7	; pointer to 'curent sample' store (within 'something')
+ ||        ADD.L1        A0,-1,A0	; tweak for BDEC
+ ||        ADD.S1X       A5,B19,A4	; add offset, pointer to 'final' output
+$C$L1_5:
+;$C$L1:
+$C$L2_5:
+;$C$L2:
+           		LDW.D2T2      *B6++[1],B5	; read 'effect' input
+           		NOP           4
+
+           		STW.D2T2      B5,*+B7[0]	; write to 'current sample'? B7 never increased...
+           		LDW.D2T2      *+B4[5],B17	; K5
+           		LDW.D2T2      *+B8[0],B5	; read previous sample back from 'effect' output???
+							; note B8: increases as loop runs
+           		NOP           3
+
+           		LDW.D2T2      *B4[0],B16	; K0
+           		MPYSP.M2      B5,B17,B18 	; Sample * K5
+           		NOP           2			;
+           		LDW.D2T2      *B4[4],B17	; K4
+           		MPYSP.M2      B16,B18,B16 	; K0 * (Sample * K5)
+           		NOP           3			;
+           		MPYSP.M2      B17,B16,B16	; K4 * (K0 * (Sample * K5))
+           		NOP           3			;
+           		STW.D2T2      B16,*B8++[1]	; write audio 'effect' output
+
+           		LDW.D2T2      *B4[6],B16	; K6, why bother? <= delay slots = 4 
+           		LDW.D2T2      *B4[0],B17	; K0
+           		NOP           4			;
+           		SUBSP.L2      B9,B17,B16	; 1.0 - K0 (delay slots = 1)
+ ||        		MPYSP.M2      B5,B16,B5 	; Sample * K6
+           		NOP           2			;
+           		LDW.D2T2      *+B4[4],B17	; K4
+           		MPYSP.M2      B16,B5,B5		;(1.0 - K0) * (Sample * K6)
+           		NOP           2			;
+           		LDW.D1T1      *A4[0],A3		; read previous sample back from 'final' output
+							; note: A4 increases as loop runs
+           		MPYSP.M2      B17,B5,B5		; K4 * ((1.0 - K0) * (Sample * K6))
+           		NOP           2			;
+    [ A0]  		BDEC.S1       $C$L1_5,A0	; LOOP 15 times back to $C$L1 (delay slots = 5)
+           		ADDSP.L2X     B5,A3,B5		; combine previous 'final' and curent sample
+           		NOP           3			;
+           		STW.D1T2      B5,*A4++[1]	; write audio to 'final' output
+
+           	ADD.L2        B0,-1,B0		; decrement outer loop counter
+ ||        	MVK.S1        16,A3		; reload 16 loops
+ ||        	ADDK.S2       64,B19		; 2nd outer loop adds 64byte offset 'effect' output buffer?
+ ||        	MV.D2         B21,B6		; update pointer to input audio (same audio repeated?)
+
+           	ADD.L1        A3,-1,A0		; tweak for BDEC
+ ||        	ADD.L2        B19,B20,B8	; increase/update pointer to output audio block
+ || [ B0]  	B.S1          $C$L1_5		; outer loop back to $C$L1
+    [!B0]  	BNOP.S2       B3,4		; add delay to compensate for delay slots
+
  ||        ADD.L1X       A5,B19,A4
-           NOP           
-           NOP           
-           NOP           
-           NOP           
-           NOP           
-           NOP           
+           ;NOP           
+           ;NOP           
+           ;NOP           
+           ;NOP           
+           ;NOP           
+           ;NOP           
 
 	.def __call_stub
 	.align 32
@@ -250,23 +254,23 @@
  ||        LDDW.D2T1     *++B15[1],A7:A6
            LDW.D2T1      *++B15[2],A2
            NOP           4
-           NOP           
+           ;NOP           
 
 	.def Dll_LineSel
 	.align 32
 Dll_LineSel:
            BNOP.S2       B3,1
            MVK.L1        4,A0
- ||        MVK.S2        0x0130,B0
+ ||        MVKL.S2       LineSel,B0
 	; 00000264  00000909 R_C6000_ABS_L16        80000130   LineSel + 0
-           MVK.S1        0x0000,A1
+           MVKL.S1       effectTypeImageInfo,A1
 	; 00000268  00000809 R_C6000_ABS_L16        80000000   effectTypeImageInfo + 0
- ||        MVKH.S2       0x80000000,B0
+ ||        MVKH.S2       LineSel,B0
 	; 0000026c  0000090a R_C6000_ABS_H16        80000130   LineSel + 0
  ||        STB.D1T1      A0,*A4[0]
            STW.D1T2      B0,*A4[1]
- ||        MVKH.S1       0x80000000,A1
-	; 00000274  0000080a R_C6000_ABS_H16        80000000   effectTypeImageInfo + 0
+ ||        MVKH.S1       effectTypeImageInfo,A1
+	; 00000274  0000080a R_C6000_ABS_H16   80000000   effectTypeImageInfo + 0
            STW.D1T1      A1,*+A4[3]
            ;.fphead       n, l, W, BU, nobr, nosat, 0010001b
 
@@ -286,8 +290,8 @@
  ||        LDDW.D2T1     *++B15[1],A15:A14
            LDW.D2T2      *++B15[2],B14
            NOP           4
-           NOP           
-           NOP           
+           ;NOP           
+           ;NOP           
            ;.fphead       n, l, W, BU, nobr, nosat, 0001111b
 
 	.def __push_rts
@@ -305,8 +309,8 @@
            STDW.D2T2     B11:B10,*B15--[1]
            STDW.D2T1     A11:A10,*B15--[1]
            STDW.D2T2     B3:B2,*B15--[1]
-           NOP           
-           NOP           
+           ;NOP           
+           ;NOP           
            ;.fphead       n, l, W, BU, nobr, nosat, 0011100b
 	.sect ".const"
 
@@ -318,7 +322,7 @@
 effectTypeImageInfo:
            .word 0x00000017
            .word 0x00000020
-           .word 0x80000210
+           .word _picTotalDisplay_LineSel
 	; 80000008  00000201 R_C6000_ABS32          80000210   _picTotalDisplay_LineSel + 0
            .word 0x00000000
            .word 0x00000000
@@ -408,7 +412,7 @@
            .word 0x00000000
            .word 0x00000000
            .word 0x00000000
-           .word 0x00000000
+           .word Fx_SFX_LineSel_onf
 	; 8000014c  00000701 R_C6000_ABS32          00000000   Fx_SFX_LineSel_onf + 0
            .word 0x00000000
            .word 0x00000000
@@ -423,23 +427,23 @@
            .word 0x00000000
            .word 0x00000001
            .word 0x00000000
-           .word 0x00000100
+           .word Fx_SFX_LineSel_init
 	; 80000184  00000601 R_C6000_ABS32          00000100   Fx_SFX_LineSel_init + 0
-           .word 0x00000140
+           .word Fx_SFX_LineSel
 	; 80000188  00000501 R_C6000_ABS32          00000140   Fx_SFX_LineSel + 0
            .word 0x00000000
            .word 0x00000000
            .word 0x00000000
            .word 0x00000000
            .word 0x00000000
-           .word 0x20584645
-           .word 0x004c564c
+           .word 0x2d786645			; Changed to 'Efx-Lvl' to confirm effect loaded
+           .word 0x006c764c
            .word 0x00000000
            .word 0x00000096
            .word 0x00000064
            .word 0x00000096
            .word 0x00000000
-           .word 0x00000064
+           .word Fx_SFX_LineSel_EfxLvl_edit
 	; 800001bc  00000401 R_C6000_ABS32          00000064   Fx_SFX_LineSel_EfxLvl_edit + 0
            .word 0x00000000
            .word 0x00000000
@@ -454,7 +458,7 @@
            .word 0x00000064
            .word 0x00000096
            .word 0x00000000
-           .word 0x000000b0
+           .word Fx_SFX_LineSel_OutLvl_edit
 	; 800001f4  00000301 R_C6000_ABS32          000000b0   Fx_SFX_LineSel_OutLvl_edit + 0
            .word 0x00000000
            .word 0x00000000
